packages/framework/src/Component/Router/NormalizeUrlTrailingSlashSubscriber.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Router;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+
+class NormalizeUrlTrailingSlashSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Router\CurrentDomainRouter
+     */
+    protected $router;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Router\CurrentDomainRouter $router
+     */
+    public function __construct(CurrentDomainRouter $router)
+    {
+        $this->router = $router;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            KernelEvents::EXCEPTION => ['onKernelException'],
+        ];
+    }
+
+    /**
+     * @param \Symfony\Component\HttpKernel\Event\ExceptionEvent $event
+     */
+    public function onKernelException(ExceptionEvent $event)
+    {
+        if ($event->getThrowable() instanceof NotFoundHttpException) {
+            $pathInfo = $event->getRequest()->getPathInfo();
+
+            if (substr($pathInfo, -1) === '/') {
+                $pathInfo = rtrim($pathInfo, '/');
+            } else {
+                $pathInfo .= '/';
+            }
+
+            $this->redirectIfPathExists($pathInfo, $event);
+        }
+    }
+
+    /**
+     * @param string $newPath
+     * @param \Symfony\Component\HttpKernel\Event\ExceptionEvent $event
+     */
+    protected function redirectIfPathExists(string $newPath, ExceptionEvent $event): void
+    {
+        try {
+            $this->router->match($newPath);
+
+            $pathInfo = $event->getRequest()->getPathInfo();
+            $fullPath = $event->getRequest()->getRequestUri();
+            $pathToRedirect = str_replace($pathInfo, $newPath, $fullPath);
```

Comment:
This is magic and It does not work with path like `www` or `http`...

Reply:
or with same string in GET parameter

Reply:
I have been testing this on full url with filter parameters, can you be more specific please?

Reply:
I did not test it, however I guess this model situation:
```php
$fullPath = 'https://example.com/com/?foo=example.com/bar';
$pathInfo = 'com/';
$newPath = 'com';
$pathToRedirect = 'https://example.com/com?foo=example.combar'; # !!!
```
It can by solved by replace only first occur, but:
```php
$fullPath = 'https://example.com/example';
$pathInfo = 'example';
$newPath = 'example/';
$pathToRedirect = 'https://example/com/example/'; # !!!
```
Maybe it can be fixed by ignoring protocol and domain, but It can be obstacle for individual routing, when pathinfo contains some prefix like `https://example.com/en/order`. 

It is reason, why I prefer for generation url by router

Reply:
I have updated it to be more accurate